common/src/main/java/com/alibaba/nacos/common/http/HttpClientBeanFactory.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 1999-2018 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.common.http;
+
+import com.alibaba.nacos.common.http.client.NacosAsyncRestTemplate;
+import com.alibaba.nacos.common.http.client.NacosRestTemplate;
+import com.alibaba.nacos.common.utils.ExceptionUtil;
+import com.alibaba.nacos.common.utils.ThreadUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Create a rest template
+ * to ensure that each custom client config and rest template are in one-to-one correspondence
+ *
+ * @author mai.jh
+ * @date 2020/6/16
+ */
+@SuppressWarnings("all")
+public final class HttpClientBeanFactory {
+
+    private static final Logger logger = LoggerFactory.getLogger(HttpClientManager.class);
+
+    private static final int TIMEOUT = Integer.getInteger("nacos.http.timeout", 5000);
+
+    private static HttpClientConfig HTTP_CLIENT_CONFIG = HttpClientConfig.builder()
+        .setConTimeOutMillis(TIMEOUT).setReadTimeOutMillis(TIMEOUT >> 1).build();
+
+    private static final Map<String, NacosRestTemplate> singletonRest = new HashMap<String, NacosRestTemplate>(10);
+    private static final Map<String, NacosAsyncRestTemplate> singletonAsyncRest = new HashMap<String, NacosAsyncRestTemplate>(10);
+
+    private static final AtomicBoolean alreadyShutdown = new AtomicBoolean(false);
+
+    static {
+        ThreadUtils.addShutdownHook(new Runnable() {
+            @Override
+            public void run() {
+                shutdown();
+            }
+        });
+    }
+
+    public static NacosRestTemplate getNacosRestTemplate() {
+        HttpClientFactory httpClientFactory = new DefaultHttpClientConfig();
+        return getNacosRestTemplate(httpClientFactory);
```

Comment:
merge these line to single line to remove tmp variable.

like `return getNacosRestTemplate(new DefaultHttpClientConfig());`